In [2]:
import pandas as pd
import matplotlib as plt

Lisons le fichiers 'countries.csv', et affichons les cinq premières lignes:

In [3]:
countries = pd.read_csv('countries.csv')
countries.head()

country     native_name iso2 iso3  population       area    capital  \
0     Afghanistan       افغانستان   AF  AFG  26023100.0   652230.0      Kabul   
1         Albania       Shqipëria   AL  ALB   2895947.0    28748.0     Tirana   
2         Algeria         الجزائر   DZ  DZA  38700000.0  2381741.0    Algiers   
3  American Samoa  American Samoa   AS  ASM     55519.0      199.0  Pago Pago   
4          Angola          Angola   AO  AGO  24383301.0  1246700.0     Luanda   

   capital_lat  capital_lng                     region continent  
0    34.526011    69.177684  Southern and Central Asia      Asia  
1    41.326873    19.818791            Southern Europe    Europe  
2    36.775361     3.060188            Northern Africa    Africa  
3   -14.275479  -170.704830                  Polynesia   Oceania  
4    -8.827270    13.243951             Central Africa    Africa

Taille du data sat et noms des colonnes: 

In [4]:
countries.shape, countries.columns

((214, 11),
 Index(['country', 'native_name', 'iso2', 'iso3', 'population', 'area',
        'capital', 'capital_lat', 'capital_lng', 'region', 'continent'],
       dtype='object'))

On a 214 lignes donc 214 pays avec leur données pour 8 colonnes différentes.\
Il serait alors cohérent d'indexer nos datas par le pays correspondant:

In [9]:
countries=countries.set_index('country')
countries.tail() #affichons la fin pour changer 

native_name iso2 iso3  population      area   capital  \
country                                                                         
Wallis and Futuna  Wallis et Futuna   WF  WLF     13135.0     142.0  Mata-Utu   
Western Sahara      الصحراء الغربية   EH  ESH    586000.0  266000.0  El Aaiún   
Yemen                       اليَمَن   YE  YEM  25956000.0  527968.0    Sana'a   
Zambia                       Zambia   ZM  ZMB  15023315.0  752612.0    Lusaka   
Zimbabwe                   Zimbabwe   ZW  ZWE  13061239.0  390757.0    Harare   

                   capital_lat  capital_lng           region continent  
country                                                                 
Wallis and Futuna   -13.282042  -176.174022        Polynesia   Oceania  
Western Sahara       27.154512   -13.195392  Northern Africa    Africa  
Yemen                15.353857    44.205884      Middle East      Asia  
Zambia              -15.416449    28.282154   Eastern Africa    Europe  
Zimbabwe            -17.831773    31.045686   Eastern Africa    Africa

Données statistiques de chaque colonnes :

In [7]:
for colonne in countries.columns:
    print(countries[colonne].describe(),'\n')

count             214
unique            214
top       Afghanistan
freq                1
Name: country, dtype: object 

count           213
unique          213
top       افغانستان
freq              1
Name: native_name, dtype: object 

count     213
unique    212
top        GB
freq        2
Name: iso2, dtype: object 

count     214
unique    213
top       GBR
freq        2
Name: iso3, dtype: object 

count    2.100000e+02
mean     3.365144e+07
std      1.330461e+08
min      3.000000e+01
25%      7.572605e+05
50%      6.315500e+06
75%      2.268378e+07
max      1.367110e+09
Name: population, dtype: float64 

count    2.070000e+02
mean     6.537604e+05
std      1.866909e+06
min      2.020000e+00
25%      1.306600e+04
50%      1.098840e+05
75%      4.817710e+05
max      1.712444e+07
Name: area, dtype: float64 

count          212
unique         211
top       Kingston
freq             2
Name: capital, dtype: object 

count    212.000000
mean      18.812330
std       26.590545
min      -54.28

Encore une fois les types de chaque colonne sont cohérents.
Pour chaque colonne, on a le nom de la colonne, le type des éléments qu'elle contient, le nombre d'éléments (count), le nombre d'élements différents (unique), l'élément le plus fréquent (top, pour les éléments de type objet) et sa fréquence d'apparition, et pour les colonnes dont le type est des float 64, on a le nombre de valeurs non nulles (count, les valeurs NaN ne sont pas comptées), la moyenne, l'écart-type, le minimum, les quartiles, le maximum. 